static/docs/commands-reference/commit.md

```diff
@@ -2,40 +2,212 @@
+  `dvc commit` when ready.
+
 ## Options
 
+* `-d`, `--with-deps` - determines the files to commit by searching backwards in
+  the pipeline from the named stage(s). The only files which will be committed
+  are associated with the named stage, and the stages which execute earlier in
+  the pipeline.
+
+* `-R`, `--recursive` - the `targets` value is expected to be a directory path.
+  With this option, `dvc commit` determines the files to commit by searching the
+  named directory, and its subdirectories, for DVC files for which to commit
+  data. Along with providing a `target`, or `target` along with `--with-deps`,
+  it is yet another way to limit the scope of DVC files to upload.
+
+* `-f`, `--force` - commit data even if checksums for dependencies or outputs
+  did not change.
+
 * `-h`, `--help` - prints the usage/help message, and exit.
 
-* `-q`, `--quiet` - does not write anything to standard output. Exit with 0 if
-  no problems arise, otherwise 1.
+* `-q`, `--quiet` - do not write anything to standard output. Exit with 0 if all
+  stages are up to date or if all stages are successfully rerun, otherwise exit
+  with 1.
+
+* `-v`, `--verbose` - displays detailed tracing information from executing the
+  `dvc add` command.
 
-* `-v`, `--verbose` - displays detailed tracing information.
 
 ## Examples
 
+Let's start with the example workspace used in the
+[Getting Started](/doc/get-started) tutorial.  Setup requires these steps:
+
+```dvc
+    $ git clone https://github.com/iterative/example-get-started
+    $ cd example-get-started
+    $ dvc pull
+```
+
+The workspace in the Git repository is preconfigured with a DVC remote containing
+precomputed data we can use in the following examples.
+
+## Example: Editing a data file
+
+Sometimes we want to edit a data or configuration file and rerun the pipeline to
+see what difference it makes.  To replace a data file we'd use the `dvc remove`
+command, but to edit it we use `dvc unprotect`.
+
+```dvc
+    $ ls -l data/
+    total 74072
+    -rw-r--r--  2 david  admin  37916850 Apr 11 17:03 data.xml
+
+    $ dvc unprotect data/data.xml
+    [##############################] 100% Unprotecting 'data/data.xml'
+
+    $ ls -l data/
+    total 74072
+    -rw-r--r--  1 david  admin  37916850 Apr 11 17:06 data.xml
+```
+
+The `1` in link count here indicates DVC has unprotected the file.  For other
+link types this will appear in different ways.  At this point we can safely edit
+the data file.
+
+After editing the file we can use `dvc commit` to update the changes.
+
+```dvc
+    $ dvc commit
+    ...
+    outputs ['data/data.xml'] of 'data/data.xml.dvc' changed. Are you sure you commit it? [y/n] y
+    ...
+    dependencies ['data/data.xml'] of 'prepare.dvc' changed. Are you sure you commit it? [y/n] y
+```
+
+Because this file is referenced from two DVC files, we are queried twice.
+
+## Example: Running the pipeline without committing data changes
```

Comment:
I would make this example first. This case is more important than `dvc add`. For `dvc add` there are other options at least.

Reply:
Yes I had thought the same